tests/decode_images.rs

```diff
@@ -183,6 +187,43 @@ fn issue_69() {
     test_image_sum_u16("issue_69_packbits.tiff", ColorType::Gray(16), 1015486);
 }
 
+/// Relies on the "generate_new_hash_for_scanimage_metadata_parser" 'test' to
+/// generate the correct hash value to be compared.
+#[test]
+fn test_si_header_metadata_parser() {
+    let filename = "ScanImageBigTIFFv57.tif";
+    let path = PathBuf::from(TEST_IMAGE_DIR).join(BIGTIFF_DIR).join(filename);
+    let file = File::open(path).unwrap();
+    let decoder = Decoder::new(file).unwrap().with_scanimage();
+    let meta = decoder.scanimage_metadata().unwrap();
+    let mut hasher_roi = DefaultHasher::new();
+    let mut hasher_frame = DefaultHasher::new();
+    let _ = meta.roi_group_data.hash(&mut hasher_roi);
+    let _ = meta.nonvar_frame_data.hash(&mut hasher_frame);
+    assert_eq!(format!("{:x}", hasher_roi.finish()), "f721c512d06e4a62");
+    assert_eq!(format!("{:x}", hasher_frame.finish()), "f62a98e18c564b82");
+}
+
+/// Generates a new hash value for the metadata in the ScanImage demo file.
+/// It should only be run if the parsing function was knowingly changed which
+/// changes the type of output of the SI metadata. The current hashes are
+/// "f721c512d06e4a62" for the ROI and "f62a98e18c564b82" for the Frame data.
+#[test]
+#[ignore]
+fn generate_new_hash_for_scanimage_metadata_parser() {
+    let filename = "ScanImageBigTIFFv57.tif";
+    let path = PathBuf::from(TEST_IMAGE_DIR).join(BIGTIFF_DIR).join(filename);
+    let file = File::open(path).unwrap();
+    let decoder = Decoder::new(file).unwrap();
+    let meta = decoder.scanimage_metadata().unwrap();
+    let mut hasher_roi = DefaultHasher::new();
+    let mut hasher_frame = DefaultHasher::new();
+    let _ = meta.roi_group_data.hash(&mut hasher_roi);
+    let _ = meta.nonvar_frame_data.hash(&mut hasher_frame);
```

Comment:
Would you move the calculation into a common method to ensure this skipped test does not silently diverge?